src/main/scala/com/microsoft/hyperspace/index/dataskipping/DataSkippingIndex.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (2021) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index.dataskipping
+
+import org.apache.spark.sql._
+import org.apache.spark.sql.catalyst.expressions.NamedExpression
+import org.apache.spark.sql.functions.input_file_name
+
+import com.microsoft.hyperspace.index._
+import com.microsoft.hyperspace.index.dataskipping.sketch.Sketch
+
+case class DataSkippingIndex(
+    sketches: Seq[Sketch],
+    sketchOffsets: Seq[Int],
+    override val properties: Map[String, String] = Map.empty)
```

Comment:
This is NOT overriding the property with an empty map. `override` means that `val properties` is overriding (implementing) the interface. The default value means that you don't have to (you can if you want to) specify the value for `properties` when creating an instance of `DataSkippingIndex`. It is used here because `properties` merely exists to satisfy the interface and not the interest of `DataSkippingIndex`.